Storage/src/Connection/RetryTrait.php

```diff
@@ -173,13 +179,24 @@ private function retryDeciderFunction(
         $isIdempotent,
         $preconditionNeeded,
         $preconditionSupplied,
-        $maxRetries
+        $maxRetries,
+        $retryStrategy
     ) {
         // No retry if maxRetries reached
         if ($maxRetries <= $currentAttempt) {
             return false;
         }
 
+        switch ($retryStrategy) {
```

Comment:
This logic is not quite correct. `RETRY_STRATEGY_NEVER` should definitely return false here; however with `RETRY_STRATEGY_ALWAYS` you still need to check whether the error code is retryable.

Equivalent logic in Go is here: https://github.com/googleapis/google-cloud-go/blob/storage/v1.28.1/storage/invoke.go#L50

Reply:
Ohh, Okay.

It seems I had some misunderstanding regarding this case earlier. Fixed it in [a47e376](https://github.com/googleapis/google-cloud-php/pull/5731/commits/a47e3761927e63746dafff9048bc0c8b17b74bb0).

Thanks.

Reply:
cc @saranshdhingra

Reply:
Just want to confirm @tritone 
If this is the case then there is no difference in the "always" and the "idempotent" case.

As per my understanding, we wanted to give user the ability to have these options:

- I know this isn't an idempotent op, but I want to retry this if it fails. -> retry always.
- I don't want you to retry any op if there is an error, I'll take care of that logic myself. -> retry never.
- If you folks think the op should be retried, please do so. -> Idempotent case(default)

Reply:
Confirmed offline. The current logic is fine. We'll go ahead with this.